.husky/git-commit-checks.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/python3
+import subprocess
+import sys
+import os
+from collections import defaultdict
+
+def get_list_of_staged_files():
+    staged_files = subprocess.run("git diff --name-only --cached", shell = True, capture_output = True,
+                                  text = True).stdout
+    return staged_files.rstrip("\n").split("\n")
+
+def group_files_by_extension(list_of_staged_files):
+    staged_files_grouped_by_extension = defaultdict(list)
+
+    for file in list_of_staged_files:
+        if "." in file:
+            extension_name = file[file.find(".") + 1:]
+            staged_files_grouped_by_extension[extension_name].append(file)
+
+    return staged_files_grouped_by_extension
+
+def is_scala_file_controller_or_spec_test(scala_files):
```

Comment:
Yes, as it's checking if the scala file is in the controller directory or a test as those are the only ones we are concerned about when it comes to running tests, but I've now realised that this is being run in the `run_style_checkers` instead of the `run_tests` and there are also non-controllers that have tests, so I will correct this. Thanks